components/FilterLeadsCard/FilterLeadsCard.tsx

```diff
@@ -0,0 +1,236 @@
+'use-client';
+
+import { LEADS_DATA } from '@/utils/constants/leadslist-constant';
+import { useState } from 'react';
+import { CustomMultiSelect } from '../CustomMultiSelect/custom-multi-select';
+import 'react-dates/initialize';
+import 'react-dates/lib/css/_datepicker.css';
+import { DateRangePicker } from 'react-dates';
+import { Moment } from 'moment';
+
+interface statusState {
+  hot: boolean;
+  warm: boolean;
+  cold: boolean;
+}
+
+interface FilterLeadsCardProps {
+  setFilterCardOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  onFilterData: (data: any) => void;
+}
+
+const FilterLeadsCard: React.FC<FilterLeadsCardProps> = ({
+  onFilterData,
+  setFilterCardOpen,
+}) => {
+  //! State for date range
+  const [startDate, setStartDate] = useState<Moment | null>(null);
+  const [endDate, setEndDate] = useState<Moment | null>(null);
+  const [focusedInput, setFocusedInput] = useState<'startDate' | 'endDate' | null>(null);
+
+  //! State for status
+  const [status, setStatus] = useState<statusState>({
+    hot: false,
+    warm: false,
+    cold: false,
+  });
+
+  const handleCheckboxChange = (statusName: keyof statusState) => {
+    setStatus((prevStatus) => ({
+      ...prevStatus,
+      [statusName]: !prevStatus[statusName],
+    }));
+  };
+
+  //! State for Created by
+  const [selectedCreatedBy, setCreatedBy] = useState<string | null>(null);
+  const [selectedAssignee, setAssignee] = useState<string | null>(null);
+
+  const createdByUsersList: { value: string; label: string }[] = [];
```

Comment:
Move these constants i.e. `createdByUsersList` & `assigneeUsersList` dummy constants into `constants` folder. The code will looks like: 
```javascript
const createdByUsersList = LEADS_DATA.map((item) => ({
  value: item.assignedByName,
  label: item.assignedByName,
});

const assigneeUsersList = LEADS_DATA.map((item) => ({
  value: item.assignedToName,
  label: item.assignedToName,
});
```